File: news-api/src/main/java/com/news/search/infrastructure/KakaoWebRepositoryImpl.java
package com.news.search.infrastructure;

import com.news.feign.KakaoClient;
import com.news.model.Document;
import com.news.model.KakaoWebResponse;
import com.news.search.service.port.WebRepository;
import com.news.search.service.response.PageQueryResult;
import com.news.search.service.response.SearchQueryResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

import java.util.List;

@RequiredArgsConstructor
@Repository
public class KakaoWebRepositoryImpl implements WebRepository {
    private final KakaoClient kakaoClient;

    @Override
    public PageQueryResult<SearchQueryResponse> search(String query, int page, int size) {
        KakaoWebResponse response = kakaoClient.search(query, page, size);
        List<SearchQueryResponse> responses = response.documents().stream()
                .map(this::convertToSearchResponse)
                .toList();
        return new PageQueryResult<>(page, size, response.meta().totalCount(), responses);
    }

    private SearchQueryResponse convertToSearchResponse(Document document){
        return SearchQueryResponse.builder()
                .title(document.title())
                .link(document.url())
                .description(document.contents())
                .build();
    }
}

File: news-api/src/main/kotlin/com/search/websearch/infrastructure/KakaoWebRepositoryImpl.kt
package com.search.websearch.infrastructure

import com.search.feign.KakaoClient
import com.search.model.Document
import com.search.websearch.service.port.WebRepository
import com.search.websearch.infrastructure.result.WebSearchPageResult
import com.search.websearch.infrastructure.result.WebSearchResult
import org.springframework.stereotype.Repository

@Repository
class KakaoWebRepositoryImpl(
        val kakaoClient: KakaoClient
) : WebRepository{
    override fun search(query: String, page: Int, size: Int): WebSearchPageResult<WebSearchResult> {
        val searchResult = kakaoClient.search(query,page, size)
        val contents = searchResult.documents.map { toWebSearchResult(it) }
        return WebSearchPageResult(page, size, searchResult.meta.totalCount, contents)
    }

    private fun toWebSearchResult(document: Document): WebSearchResult {
        return WebSearchResult(
                title = document.title,
                link = document.url,
                description = document.contents
        )
    }
}